src/backend/llvm.py

```diff
@@ -46,7 +46,10 @@ def generate(self, bril_prog):
     def gen_type(self, type):
         if isinstance(type, dict):
             if "ptr" in type:
-                return self.gen_type(type["ptr"]).as_pointer()
+                type_obj = self.gen_type(type["ptr"]).as_pointer()
```

Comment:
as_pointer take addrspace as argument

Reply:
I thought it's cleaner this way, because there is only a single call to gen_type. Should I change it?

Reply:
Done